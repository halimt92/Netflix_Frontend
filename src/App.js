import React, { useEffect, useState } from "react";
import Routes from "./routes/Routes";
import axios from "./axios/axios";
import { useDispatch } from "react-redux";
import { LastLocationProvider } from "react-router-last-location";
import Navbar from "./components/navbar/Navbar";
import Footer from "./components/footer/Footer";
import { BrowserRouter } from "react-router-dom";
import useLocalStorage from "./hooks/uselocalstorage";
import { setupToken } from "./redux/actions/user-actions/userAction";
import { fetchFavMovies } from "./redux/actions/fav-movies-actions/favMoviesAction";
//just to debug and check the userReducer state
import { useSelector } from "react-redux";

export const TOKEN_STORAGE_ID = "netflix-token";

const App = () => {
  const [token, setToken] = useLocalStorage(TOKEN_STORAGE_ID);
  const dispatch = useDispatch();
  const user = useSelector((state) => state.user);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    dispatch(axios());
    if (token) {
      async function setUser() {
        await dispatch(setupToken(token));
        await dispatch(fetchFavMovies());
      }
      setUser();
      setLoading(false);
    }
    setLoading(false);
  }, []);

  if (loading) {
    return <div />;
  } else {
    return (
      <BrowserRouter>
        <LastLocationProvider>
          <Navbar />
          <Routes />
          <Footer />
        </LastLocationProvider>
      </BrowserRouter>
    );
  }
};

export default App;
