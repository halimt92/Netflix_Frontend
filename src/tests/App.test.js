import React from 'react';
import { shallow } from 'enzyme';
import App from '../container/App';
import BillboardRow from '../components/billboard-row/BillboardRow';
import RowContainer from '../container/row-container/RowContainer';

describe('App', () => {
  it('should render a <BillboardRow /> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(BillboardRow)).toHaveLength(1);
  });

  it('should render a <BillboardRow /> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(RowContainer)).toHaveLength(1);
  });

  it('matches App snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
