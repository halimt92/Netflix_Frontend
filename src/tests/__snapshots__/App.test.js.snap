// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aoo matches App snapshot 1`] = `ShallowWrapper {}`;

exports[`App matches App snapshot 1`] = `ShallowWrapper {}`;

exports[`Footer matches App snapshot 1`] = `ShallowWrapper {}`;
